packages/relay/src/RegistrationManager.ts

```diff
@@ -412,12 +412,19 @@ export class RegistrationManager {
     // add worker only if not already added
     const workersAdded = await this._isWorkerValid()
     const addWorkersPending = await this.txStoreManager.isActionPendingOrRecentlyMined(ServerAction.ADD_WORKER, currentBlockNumber, this.config.recentActionAvoidRepeatDistanceBlocks)
+    let skipGasEstimationForRegisterRelay = false
     if (!(workersAdded || addWorkersPending)) {
       const txHash = await this.addRelayWorker(currentBlockNumber, currentBlockTimestamp)
       transactions = transactions.concat(txHash)
+      skipGasEstimationForRegisterRelay = true
     }
     const registerMethod = await this.contractInteractor.getRegisterRelayMethod(this.hubAddress, this.config.baseRelayFee, this.config.pctRelayFee, this.config.url)
-    const gasLimit = await this.transactionManager.attemptEstimateGas('RegisterRelay', registerMethod, this.managerAddress)
+    let gasLimit: number
```

Comment:
same is more relevant here.
just gasLeft=value if you want explicit value, or leave it as null to estimate..

Reply:
Fixed.